examples/blender/blender.py

```diff
@@ -78,6 +78,10 @@ async def worker(ctx: WorkContext, tasks):
                 )
                 raise
 
+            print(f" --- {ctx.provider_name} USAGE: {await ctx.get_usage()}")
+            print(f" --- {ctx.provider_name} STATE: {await ctx.get_state()}")
+            print(f" --- {ctx.provider_name}  COST: {await ctx.get_cost()}")
+
```

Comment:
I'm not sure we want to print it to the console in `blender.py`, at least in this form it looks rather low-level. Should the user be concerned with the provider state (expressed as a pair of states of some other component)?

Reply:
Besides, why is usage and cost always `None`?:
```
[2021-07-01T15:39:01.763+0200 INFO yapapi.summary] Task started on provider 'provider-2', task data: 10
[2021-07-01T15:41:47.293+0200 WARNING yapapi.summary] Activity failed on provider 'provider-2'; reason: Command '{'deploy': {}}' failed on provider; message: 'Local service error: Transfer error: Downloaded VM image is corrupted: calculated hash 4ea6048dc19f2b737db5bf18722a979d1e9b29ff7776ec4614923c17 differs from the expected one 9a3b5d67b0b27746283cb5f287c13eab1beaa12d92a9f536b747c7ae'
[2021-07-01T15:41:49.121+0200 INFO yapapi.summary] Terminated agreement with provider-2
[2021-07-01T15:41:49.174+0200 INFO yapapi.summary] Received proposals from 2 providers so far
[2021-07-01T15:41:50.414+0200 INFO yapapi.summary] Accepted invoice from 'provider-2', amount: 0.06861949319805555
[2021-07-01T15:41:51.127+0200 INFO yapapi.summary] Agreement proposed to provider 'provider-2'
[2021-07-01T15:41:51.180+0200 INFO yapapi.summary] Agreement confirmed by provider 'provider-2'
[2021-07-01T15:41:52.693+0200 INFO yapapi.summary] Task started on provider 'provider-2', task data: 10
Task computed: Task(id=1, data=0), result: output_0.png, time: 0:02:57.677019
 --- provider-1 USAGE: {'current_usage': None, 'timestamp': 1625146739}
 --- provider-1 STATE: {'error_message': None, 'reason': None, 'state': ['Ready', None]}
 --- provider-1  COST: None
[2021-07-01T15:41:57.348+0200 INFO yapapi.summary] Task finished by provider 'provider-1', task data: 0
[2021-07-01T15:41:57.349+0200 INFO yapapi.summary] Task started on provider 'provider-1', task data: 20
Task computed: Task(id=3, data=20), result: output_20.png, time: 0:00:09.166738
 --- provider-1 USAGE: {'current_usage': None, 'timestamp': 1625146739}
 --- provider-1 STATE: {'error_message': None, 'reason': None, 'state': ['Ready', None]}
 --- provider-1  COST: None
[2021-07-01T15:42:06.525+0200 INFO yapapi.summary] Task finished by provider 'provider-1', task data: 20
[2021-07-01T15:42:06.525+0200 INFO yapapi.summary] Task started on provider 'provider-1', task data: 30
Task computed: Task(id=4, data=30), result: output_30.png, time: 0:00:12.179292
 --- provider-1 USAGE: {'current_usage': None, 'timestamp': 1625146739}
 --- provider-1 STATE: {'error_message': None, 'reason': None, 'state': ['Ready', None]}
 --- provider-1  COST: None
[2021-07-01T15:42:18.718+0200 INFO yapapi.summary] Task finished by provider 'provider-1', task data: 30
[2021-07-01T15:42:18.719+0200 INFO yapapi.summary] Task started on provider 'provider-1', task data: 40
Task computed: Task(id=5, data=40), result: output_40.png, time: 0:00:09.194671
 --- provider-1 USAGE: {'current_usage': None, 'timestamp': 1625146739}
 --- provider-1 STATE: {'error_message': None, 'reason': None, 'state': ['Ready', None]}
 --- provider-1  COST: None
[2021-07-01T15:42:27.935+0200 INFO yapapi.summary] Task finished by provider 'provider-1', task data: 40
[2021-07-01T15:42:27.935+0200 INFO yapapi.summary] Task started on provider 'provider-1', task data: 50
Task computed: Task(id=6, data=50), result: output_50.png, time: 0:00:09.137883
 --- provider-1 USAGE: {'current_usage': None, 'timestamp': 1625146739}
 --- provider-1 STATE: {'error_message': None, 'reason': None, 'state': ['Ready', None]}
 --- provider-1  COST: None
[2021-07-01T15:42:37.084+0200 INFO yapapi.summary] Task finished by provider 'provider-1', task data: 50
Task computed: Task(id=2, data=10), result: output_10.png, time: 0:01:54.383982
 --- provider-2 USAGE: {'current_usage': None, 'timestamp': 1625146912}
 --- provider-2 STATE: {'error_message': None, 'reason': None, 'state': ['Ready', None]}
[2021-07-01T15:43:47.086+0200 INFO yapapi.summary] Computation finished in 290.3s
```

Reply:
@azawlocki 1) could you upload the logs? 2) which yagna version is this? the fix has only been made in https://github.com/golemfactory/yagna/pull/1446 and the only release that includes it (afaik) is https://github.com/golemfactory/yagna/releases/tag/pre-rel-v0.7.2-rc1

Reply:
I don't have an opinion on whether it should be included or not...

Reply:
maybe there should be a separate example ... ? or an additional CLI parameter?

Reply:
I generally found it interesting to see the usage and cost while the activities are running :) but I'm not convinced on where should those calls be presented in the examples

Reply:
@azawlocki added a CLI parameter to show those usage stats ... cannot do that for the service example easily since we don't have an easy way to parametrize the run ;p :D

Reply:
(yet)

Reply:
> the fix has only been made in golemfactory/yagna#1446 and the only release that includes it so far (afaik) is https://github.com/golemfactory/yagna/releases/tag/pre-rel-v0.7.2-rc1

@shadeofblue Thanks, I've checked with 0.7.2 now and it works fine!